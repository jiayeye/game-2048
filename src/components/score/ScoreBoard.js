function ScoreBoard( {score, highscore} ) {
    return (
        <div className="score">
            <div style={{'float':'left'}}>
                <h3>Score: {score}</h3>
            </div>
            <div style={{'float':'right'}}>
            	<h3>Highscore: {highscore}</h3>
            </div>
        </div>
    )
}

export default ScoreBoard;