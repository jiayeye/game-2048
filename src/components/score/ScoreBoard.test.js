import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import ScoreBoard from './ScoreBoard';


describe('ScoreBoard', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<ScoreBoard />, div);
	});

	it('shows score and highscore', () => {
		const tree = renderer.create(<ScoreBoard score={8} highscore={256}/>).toJSON();
		expect(tree).toMatchSnapshot();
	});

});