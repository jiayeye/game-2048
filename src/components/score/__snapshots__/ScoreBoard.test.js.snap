// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreBoard shows score and highscore 1`] = `
<div
  className="score"
>
  <div
    style={
      Object {
        "float": "left",
      }
    }
  >
    <h3>
      Score: 
      8
    </h3>
  </div>
  <div
    style={
      Object {
        "float": "right",
      }
    }
  >
    <h3>
      Highscore: 
      256
    </h3>
  </div>
</div>
`;
